blocks/rich-text/index.js

```diff
@@ -594,10 +579,10 @@ export class RichText extends Component {
 				const beforeElement = nodeListToReact( beforeNodes, createTinyMCEElement );
 				const afterElement = nodeListToReact( afterNodes, createTinyMCEElement );
 
-				this.setContent( beforeElement );
 				this.props.onSplit( beforeElement, afterElement );
 			} else {
 				event.preventDefault();
+				this.onCreateUndoLevel();
```

Comment:
Why do we create an undo level here? The behavior of `createUndoLevel` is such that we'll trigger `onChange` which in turn calls `setAttributes`. When we're at the end of a paragraph block and pressing enter (a very common writing flow), we're thus needlessly calling `setAttributes` when the content of the original paragraph hasn't changed at all (calling twice in fact, the other a separate issue).

We should either:

- Remove this line
- Document why it's needed with an inline comment
   - And ideally try to avoid setting content

Reply:
See also #7482

Reply:
Unsure why we're adding one before splitting. Seems safe to remove, perhaps with a test ensuring right history behaviour on split.